File: lib/features/cabinets/domain/cabinet.dart
/// A consulting room (mirrors backend `CabinetOut`). Plain model — no codegen.
class Cabinet {
  const Cabinet({
    required this.id,
    required this.branchId,
    required this.name,
    this.kind,
    this.isActive = true,
  });

  final String id;
  final String branchId;
  final String name;
  final String? kind;
  final bool isActive;

  factory Cabinet.fromJson(Map<String, dynamic> j) => Cabinet(
        id: j['id'] as String,
        branchId: j['branch_id'] as String,
        name: j['name'] as String,
        kind: j['kind'] as String?,
        isActive: (j['is_active'] as bool?) ?? true,
      );
}

File: backend/app/schemas/cabinet.py
"""Cabinet (consulting room) DTOs."""
from __future__ import annotations

from uuid import UUID

from pydantic import BaseModel, ConfigDict, field_validator


class CabinetCreate(BaseModel):
    branch_id: UUID
    name: str
    kind: str | None = None

    @field_validator("name")
    @classmethod
    def _name_not_blank(cls, v: str) -> str:
        v = (v or "").strip()
        if not v:
            raise ValueError("Cabinet name must not be blank")
        return v


class CabinetUpdate(BaseModel):
    name: str | None = None
    kind: str | None = None
    is_active: bool | None = None

    @field_validator("name")
    @classmethod
    def _name_not_blank(cls, v: str | None) -> str | None:
        if v is None:
            return v
        v = v.strip()
        if not v:
            raise ValueError("Cabinet name must not be blank")
        return v


class CabinetOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: UUID
    branch_id: UUID
    name: str
    kind: str | None
    is_active: bool
